ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -456,8 +457,9 @@ List<ChunkInfo> getChunksToStitch(BlobProperties stitchedBlobProperties, JSONObj
         @SuppressWarnings("ConstantConditions")
         long expirationTimeMs = RestUtils.getLongHeader(metadata, EXPIRATION_TIME_MS_KEY, true);
         verifyChunkAccountAndContainer(blobId, stitchedBlobProperties);
+        reservedMetadataBlobId = getAndVerifyReservedMetadataBlobId(metadata, reservedMetadataBlobId, blobId);
```

Comment:
nit: would having an extra variable like `prevChunkReservedMetadataBlobId` make it easier to follow? For example:

```
String reservedMetadataBlobId = null;
String prevChunkReservedMetadataBlobId = null; 

for (String signedChunkId : signedChunkIds) {
    reservedMetadataBlobId = getAndVerifyReservedMetadataBlobId(metadata, 
    prevChunkReservedMetadataBlobId, blobId);
    chunksToStitch.add(new ChunkInfo(blobId, chunkSizeBytes, expirationTimeMs, reservedMetadataBlobId));
    prevChunkReservedMetadataBlobId = reservedMetadataBlobId;
}

```

Reply:
The thing is that the line " prevChunkReservedMetadataBlobId = reservedMetadataBlobId;" will become redundant above. Because either " prevChunkReservedMetadataBlobId" is always equal to "reservedMetadataBlobId", or `getAndVerifyReservedMetadataBlobId ` throws an exception.